src/plugins/navigation/public/plugin.tsx

```diff
@@ -66,10 +87,25 @@ export class NavigationPublicPlugin
     this.coreStart = core;
     this.depsStart = depsStart;
 
-    const { unifiedSearch, cloud } = depsStart;
+    const { unifiedSearch, cloud, security } = depsStart;
     const extensions = this.topNavMenuExtensionsRegistry.getAll();
     const chrome = core.chrome as InternalChromeStart;
 
+    if (security) {
+      this.userProfileOptOut$ = security.userProfiles.userProfileLoaded$.pipe(
```

Comment:
UserProfiles isn't guaranteed to be available (not sure exactly where to link this discussion 😅 )

How should this feature behave in this case? I.e. anonymous user 

<img width="644" alt="Screenshot 2024-03-26 at 12 18 50 PM" src="https://github.com/elastic/kibana/assets/21210601/d954084e-44d5-4666-8a5b-e15aa4ea11ce">

Reply:
> UserProfiles isn't guaranteed to be available (not sure exactly where to link this discussion 😅 )

I tried to outline all cases when user doesn't have a profile (== profile id) here https://github.com/elastic/kibana/issues/175431#issuecomment-1914577548 (TL;DR: anonymous users, interactive users authenticated via HTTP proxy, non-interactive users or HTTP clients).

Reply:
Thanks @azasypkin . I think for this use case with user facing UI we can discard non-interactive users or HTTP clients.
As you can see [on this change](https://github.com/elastic/kibana/pull/179270/files#diff-1829e5605505e1b1091dc6c14a0b0b270b656bfea4f6ba6160fdf03943fa71fbR86), I disable the user profile (updating a new `enabled$` Observable) if we receive a `404` when fetching the profile. This correspond to [this logic on the server](https://github.com/elastic/kibana/blob/main/x-pack/plugins/security/server/routes/user_profile/get_current.ts#L45).